### PR TITLE
Add automation tools for generating documentation

### PR DESCRIPTION
**Pull request by Myoldmopar**
Eventually, I think these tools should produce:
 - Release notes
 - Known issues? Maybe not...
 - Output variable summary
 - Example file summaries

For now, this pull request will probably just get things kicked off, and maybe do the output variables and the example files.  They are both pretty easy.

### CODE REVIEW

**Comment by Myoldmopar** [comment]
@kbenne I'm stealing a few minutes of your time this week to nail down a plan for things such as this. :wink:

**Comment by Myoldmopar** [comment]
@kbenne So the call the other day... we decided Python was OK, and we should push for dual 2.7/3 support?  Is that right?

**Comment by kbenne** [comment]
See the email I just forwarded to you with my meeting notes.  I wrote down Python 3, but I have no horse in the race.

**Comment by kbenne** [comment]
@Myoldmopar I'd like to use this, but don't think it should be merged until it is integrated with the build process.

**Comment by Myoldmopar** [comment]
@kbenne So I didn't comment out the install lines; I left them on so that if you want to see the output, you can just checkout this branch and run configure.  It should produce the output files, or reveal a problem.

**Comment by Myoldmopar** [comment]
(run configure _with BUILD_PACKAGE_)

**Comment by Myoldmopar** [comment]
Just tested on Windows and it worked perfectly.

**Comment by Myoldmopar** [comment]
Can you exemplify so I can monkey mimic?

**Comment by Myoldmopar** [comment]
@kbenne OK, I changed it so that it is an 'add_custom_command' that depends on the energyplus target.  It works really well.  

I tried using add_custom_target, but couldn't get it to work.  I also tried adding DEPENDS into this call, but it seems when you use the TARGET declaration, it doesn't respond to additional DEPENDS.  But this should be fine anyway.  Whenever the E+ source changes, energyplus will change, and this will get rebuilt (tested this locally).  The only thing that won't happen is if you change the actual python script, it won't trigger a rebuild.  Buuuut, our packaging scripts don't re-use folders anyway, and build everything from scratch.  

Let me know if this still needs work.

File: CMakeLists.txt [hunk 1]
```diff
@@ -44,6 +44,14 @@ option( BUILD_PACKAGE "Build package" OFF )
 option( BUILD_TESTING "Build testing targets" OFF )
 option( BUILD_FORTRAN "Build Fortran stuff" OFF )
 
+# we don't need all devs to have Python installed.  There are 2 cases where we do want it
+# 1. Doing regression testing; which would be: BUILD_TESTING AND ENABLE_REGRESSION_TESTING
+# 2. Doing package building, since Python is going to be our tool for some things; thus: BUILD_PACKAGE
+
+if( ( BUILD_TESTING AND ENABLE_REGRESSION_TESTING ) OR ( BUILD_PACKAGE ) )
+  find_package(PythonInterp REQUIRED)
```
Comment (kbenne) [inline CMakeLists.txt]: Not a Pythonist but since you tell me there is a difference between Python 2 and 3, don't you think you should specify a version?

**Comment by kbenne** [comment]
@Myoldmopar also need to make sure your robots have the correct version of Python installed.

**Comment by kbenne** [comment]
@Myoldmopar please respond to the python version question, install Python on the robots you maintain, and I will reintegrate this.

**Comment by Myoldmopar** [comment]
@kbenne Robots have 2.7.x.  I think we should target that version.  Regardless of there being new versions, we have until [2020](http://www.i-programmer.info/news/216-python/7179-python-27-to-be-maintained-until-2020.html) _at least_ to continue getting support for it.

**Comment by Myoldmopar** [comment]
@kbenne e33dd807 includes a starting point for enforcing a specific version of python.  I might be missing an obvious solution, but none of the cmake finder variables I found online seemed to enforce a version.  The language coming from the finder indicates this is finding 2.7 as a _minimum_ version, not a specific version.  Which doesn't do us much good, since we specifically want the lower 2.7 version.  I don't have a machine with only 3.0 and not 2.7 set up.  That's what I really need to test this out to ensure it _won't_ configure unless it finds 2.7.  

Maybe you know of a quick better way to find a specific version...

**Comment by Myoldmopar** [comment]
@kbenne I still don't see any existing fixes on the inter-webs to actually specify a version of python.  The above seems to make sure you have a minimum, but doesn't specify an exact version.  

Looking forward, I think it either requires a hack to the python finder to respond to a specific version variable...or after the finder runs, we check the version and make sure it finds 2.7.  If it doesn't, we issue a cmake warning to have the dev manually change the executable and library paths for 2.7.x.  I figure that sounds awful enough that it'll make you respond...

**Comment by Myoldmopar** [comment]
Oh, or we just say you need to install Python 2.7.x in the package-building machines. And make a GitHub issue to improve the finder later.

File: cmake/Install.cmake [hunk 2]
```diff
@@ -136,22 +136,16 @@ install_remote(FILES "https://raw.github.com/NREL/EnergyPlusBuildSupport/v8.2.0/
 
 # TODO Remove version from file name or generate
 # These files names are stored in variables because they also appear as start menu shortcuts later.
```
Comment (Myoldmopar) [inline cmake/Install.cmake]: @kbenne It says here these variables were being used as start menu shortcuts, but I didn't see that.  I'm hoping to clean up the install a little and get rid of some of these files (since we are more public now).  Can you confirm whether I broke something here or not? 

For the record, I was able to ```make package``` with this branch on my local machine and install it (well, [almost](https://github.com/NREL/EnergyPlus/issues/4781)) and the changelog had been created and installed nicely.  

I'm open to changing the scripts to create html instead of md, csv, etc. for the installation packages.

**Comment by Myoldmopar** [comment]
@kbenne beef980 fixes the Python issue.  For the time being, it requires a 2.7 compatible interpreter.  I do have one other thing I'm working on in this branch, and that is generating the example file summary file automatically.  Shouldn't be long before I wrap that up and this branch can drop in.

**Comment by Myoldmopar** [comment]
@kbenne So beef461 now adds in the ability to generate the example file summary purely from the idf content itself.  It reads mostly which objects are present, but also does some attempts at parsing comments.  It is adolescent and both the logic it uses, and the html output it writes, could use some beautification.  However, it should be a really nice start.  I'd like to deprecate the ExampleFiles.xls file now so we don't have to manually maintain it anymore.  It replicates almost all of the information from that spreadsheet.  It doesn't have the "tested" status, but frankly, as all the files are currently being tested, those columns will be irrelevant.  And that also shouldn't be planted in front of users' faces.  

TL;DR: Less manual maintenance!

Anyway, I also merged develop in.  This should come back clean from CI and I'd like to merge it in asap so I can stop thinking about it...

@mjwitte I know you see all comments, but here's a mention just to make sure.

**Comment by mjwitte** [comment]
@Myoldmopar Where can I see a sample?  Remember there are two flavors, ExampleFiles.xls and ExampleFiles-ObjectsLink.xls  Is this intended to replace both?  Only a partial ExampleFiles.xls was distributed previously to users (without the internal test status info.).  For users, the most important part is the comment block describing the main purpose for the example file, so I hope you can get that much included (if not already).

I realize we haven't been maintaining it, but until CI tests everything with annual and reverse dday simulations, we can't say we've fully deprecated what ExampleFiles.xls was supposed to track.

**Comment by kbenne** [comment]
@mjwitte the CI can do annuals, and I think it does elect to run them on release builds.  That is to say the gears are in place, it is just a configuration option.  It seems simple enough to add a reverse design day option, if you think we should bring that back.

**Comment by mjwitte** [comment]
@kbenne We're not even reporting numeric regressions at this point at all, so . . . some check of full annuals on iteration and release build is important to make sure things run across the full spectrum of conditions without crashing or fataling out or going into max iteration never never land.  Regarding reverse ddays - until we have a paradigm shift that ensures a clean start for every environment, it's a necessary check to be sure that nothing lingers across from one environment to the next.

**Comment by Myoldmopar** [comment]
@mjwitte I will be able to flush out ObjectsLink tomorrow morning or tonight in just a few minutes.  I was able to get the description "pretty well".  I emailed you an example.  Right now it is a flat html table.  Not real pretty. But what's nice is now that this framework is in place, it's likely very little work to add some jazz to the html to allow sorting by column, etc.

Since it is flat, I repeated the column headers every 12 lines so that users know what they are looking at.

@kbenne The reverse DD logic is embedded in there; I have it available on my regression engine (maybe not from the GUI).  We can talk with @lefticus about pushing forward bringing that back as a core part of package building.

**Comment by kbenne** [comment]
@Myoldmopar your regression picks up simulations that have already been run though right?  Doesn't the reverse design day thing need to be setup upstream of your regression script?

**Comment by mjwitte** [comment]
@Myoldmopar Thanks for the sample.  The description is a good start, seems to be just the highlights block?  Would be nice to grab both the description and the highlights blocks of text, but that can be a bit lengthy.  Some files are better than others in that regard.  Did you notice that the sizing period counts are off?  And I'd suggest we don't need to keep the "compact schedules" column anymore.  Long ago there was a push to convert the example files to using compact schedules instead of other forms.  Not really important now, and the users don't care.

**Comment by Myoldmopar** [comment]
@mjwitte I can look into adding the full description.  I wasn't sure how the html table would come out, so I didn't want to make it huge.  I'll check on the sizing count.  As for the compact schedule, yeah, it's easy to remove those, or others, or add others as needed, but I didn't want to take liberties there.  We can chat about that anytime.

**Comment by Myoldmopar** [comment]
Hmmm, yeah I don't know why CI is failing.  I can build a brand new build folder and run integration tests successfully.  Will keep investigating.

Anyway, I cleaned up the scripts a bit and fixed the sizing count (and zone count).

**Comment by Myoldmopar** [comment]
Gah!  @kbenne or @lefticus, can you see anything in the files changed that would cause this to fail integration tests across the board? I can run things fine locally, just not here.  I'll try to do it directly on the CI machine in the morning I guess.

**Comment by lefticus** [comment]
I just added some code to the CI to try and parse out python errors - my documentation build is silently failing on the build bots - perhaps this could help catch what you are seeing? You could try deleting one of the builds and see if forcing a rebuild turns anything up.

**Comment by Myoldmopar** [comment]
Thanks @lefticus I deleted the Clang since it has typically been the fastest build.  Here's to hoping.

**Comment by Myoldmopar** [comment]
@lefticus; looks like it did try to catch the Python error (https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-beef2c529483962326158fc4945b2e273f9d9c53-x86_64-MacOS-10.9-clang.html).  It doesn't give much insight, but that's OK.  I think I addressed that issue anyway in 37dd3bd, but we'll see.  I'm also doing the build/run manually on the CI box just to try to replicate any problems.

**Comment by Myoldmopar** [comment]
@lefticus And a fresh build on the CI box, Debug, with BUILD_PACKAGE and BUILD_TESTING, works just fine:

```ctest -C Debug -R 1ZoneEvapCooler``` passes in 9 seconds.  If it still fails with 37dd3bd, I might request some assistance figuring it out.

**Comment by Myoldmopar** [comment]
Noooo!!! @mbadams5   How could it possibly be failing?

**Comment by Myoldmopar** [comment]
Wait, what!?  Oh, the Clang box needs some added dependencies!  @kbenne I'll chat with you later or probably tomorrow morning.

**Comment by Myoldmopar** [comment]
Hmm, really interesting.  @mbadams5, the error is that the git command I issue to get the log since the last release is failing.  The python [error messages](https://nrel.github.io/EnergyPlusBuildResults/EnergyPlus-4fab328d9714d79bdb5247ee988be585abd11317-i386-Windows-7-VisualStudio-12.html) are now captured, thanks to @lefticus.  It was running on my Windows CI machine manually just fine, but I haven't seen a successful run from within the CI framework.  Could you try to run this python script with an appropriate path to a local clone as ```argv[1]``` and see if it clarifies the actual problem?  

```
import sys
import git
repo = sys.argv[1]
g = git.Git(repo)
log_full = g.log('--oneline', '--after="2014-9-30"')
```

**Comment by Myoldmopar** [comment]
@kbenne This PR is superseded by #4818 which includes the auto-docs, No-LTO-on-Linux, and the fixes to get NSIS working.  Closing this one.
